tests/integration/exec.go

```diff
@@ -0,0 +1,190 @@
+package integration
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/types/trace"
+	"golang.org/x/sys/unix"
+)
+
+const cpuID = 0 // CPU to pin the process to
+
+// setCPU pins the current process to a specific CPU
+func setCPU() {
+	cpuMask := unix.CPUSet{}
+	cpuMask.Set(cpuID)
+	unix.SchedSetaffinity(0, &cpuMask)
+}
+
+// parseCmd parses a command string into a command and arguments
+func parseCmd(fullCmd string) (string, []string, error) {
+	pattern := `'[^']*'|\S+` // split on spaces, but not spaces inside single quotes
+	re := regexp.MustCompile(pattern)
+	vals := re.FindAllString(fullCmd, -1)
+
+	if len(vals) == 0 {
+		return "", nil, fmt.Errorf("no command specified")
+	}
+	cmd := vals[0]
+	cmd, err := exec.LookPath(cmd)
+	if err != nil {
+		return "", nil, err
+	}
+	if !filepath.IsAbs(cmd) {
+		cmd, err = filepath.Abs(cmd)
+		if err != nil {
+			return "", nil, err
+		}
+	}
+
+	args := vals
+	// remove single quotes from args, since they can confuse exec
+	for i, arg := range args {
+		args[i] = strings.Trim(arg, "'")
+	}
+
+	return cmd, args, nil
+}
+
+// handleFDs closes all fds except for stdin, stdout, stderr and the ones we need for the exec
+// It also redirects stdout to /dev/null
+func handleFDs() error {
+	// close range of fds
+	err := unix.CloseRange(3, ^uint(0), unix.CLOSE_RANGE_UNSHARE)
+	if err != nil {
+		return fmt.Errorf("close range of fds: %w", err)
+	}
+
+	// redirect stdout to /dev/null
+	null, err := os.OpenFile("/dev/null", os.O_RDWR, 0)
+	if err != nil {
+		return fmt.Errorf("open /dev/null: %w", err)
+	}
+	err = syscall.Dup2(int(null.Fd()), unix.Stdout)
+	if err != nil {
+		return fmt.Errorf("dup2: %w", err)
+	}
+	err = syscall.Close(int(null.Fd()))
+	if err != nil {
+		return fmt.Errorf("close: %w", err)
+	}
+
+	return nil
+}
+
+// waitChild waits for a child process to exit
+func waitChild(pid int, timeout time.Duration) (int, error) {
```

Comment:
This timeout ensures we don't hang but can cause other issues like:

https://github.com/aquasecurity/tracee/actions/runs/4195065537/jobs/7274176239

```
=== RUN   Test_EventFilters/container:_event:_trace_only_events_from_new_containers
    tracee.go:81: started tracee...
	executing: /usr/bin/docker [docker run -d --rm alpine]
Unable to find image 'alpine:latest' locally
latest: Pulling from library/alpine
63b65145d645: Pulling fs layer
make: *** wait: No child processes.  Stop.
make: *** Waiting for unfinished jobs....
make: *** wait: No child processes.  Stop.
Error: Process completed with exit code 143.
```

Reply:
Fixed. Now each exec has up to 10s to finish or will be killed.